docs/Android.md

```diff
@@ -43,33 +43,10 @@ To follow along with this guide, you'll need:
 
 ## "Hello, world" on Android
 
-### 1. Downloading (or building) the Swift Android stdlib dependencies
+### 1. Building the Swift stdlib for Android
 
-You may have noticed that, in order to build the Swift stdlib for Linux, you
-needed to `apt-get install libicu-dev icu-devtools`. Similarly, building
-the Swift stdlib for Android requires the libiconv and libicu libraries.
-However, you'll need versions of these libraries that work on Android devices.
-
-The steps are as follows:
-
-1. Ensure you have `curl`, `autoconf`, `automake`, `libtool`, and
-   `git` installed.
-2. Clone the [SwiftAndroid/libiconv-libicu-android](https://github.com/SwiftAndroid/libiconv-libicu-android)
-   project. From the command-line, run the following command:
-   `git clone https://github.com/SwiftAndroid/libiconv-libicu-android.git`.
-3. From the command-line, run `which ndk-build`. Confirm that the path to
-   the `ndk-build` executable in the Android NDK you downloaded is displayed.
-   If not, you may need to add the Android NDK directory to your `PATH`.
-4. Change directories into `libiconv-libicu-android`: `cd libiconv-libicu-android`
-5. Run the Swift build script: `./build-swift.sh`
-6. Confirm that the various `libicuXYZswift.so` libraries are located in the
-   `armeabi-v7a` directory.
```

Comment:
We may need this back if we ever add instructions for cross-compiling swift-corelibs-foundation for Android as in #38441, since that repo still requires libicu, but in the meantime this libicu cross-compilation is no longer needed and we can always resurrect this doc from git if we want it later.